web/Controllers/PropertyController.cs

```diff
@@ -123,4 +124,21 @@ public ActionResult AddNewProperty_Cancel()
         // Go to View Properties page
         return RedirectToAction("ViewProperties", "Landlord");
     }
-}
+
+    [HttpGet]
+    [Route("/property/view")]
+    public ActionResult ViewProperty(int propertyId)
+    {
+        PropertyDbModel? model = _propertyService.GetPropertyByPropertyId(propertyId);
+        if (model == null)
+        {
+            _logger.LogWarning("View property did not return the property as expected");
+            return RedirectToAction(nameof(LandlordController.ViewProperties), "Landlord");

```

Comment:
Could we throw a 'not found' error instead? We shouldn't really get into this situation anyway (there should only be links to view properties which actually exist), but if a user were to manually type an ID in the URL I think it would be reasonable for them to see an error page

Reply:
ok done